app/models/patient.rb

```diff
@@ -740,23 +740,59 @@ def self.order_by_risk(asc: true)
   end
 
   # Check for potential duplicate records. Duplicate criteria is as follows:
-  # - matching first name, last name, sex, and DoB
+  # - matching values of first name and last name. Optionally sex and DoB can also match for more detailed messaging
+  # e.g. Jon Smith M null would be a duplicate of Jon Smith M 1/1/2000
   # OR
   # - matching state/local id
-  def self.duplicate_data(first_name, last_name, sex, date_of_birth, user_defined_id_statelocal)
-    dup_info = where('first_name = ?', first_name)
-               .where('last_name = ?', last_name)
-               .where('sex = ?', sex)
-               .where('date_of_birth = ?', date_of_birth)
+  def self.duplicate_data_detection(patient)
+    # Track which matches have occurred
+    duplicate_field_data = []
 
-    dup_statelocal_id = where('user_defined_id_statelocal = ?', user_defined_id_statelocal&.to_s&.strip)
+    # check for a duplicate state/local id
+    if patient[:user_defined_id_statelocal].present?
+      dup_statelocal_id = where('trim(user_defined_id_statelocal) = ?', patient[:user_defined_id_statelocal]&.to_s&.strip)
+    end
+    duplicate_field_data << { count: dup_statelocal_id.size, fields: ['State/Local ID'] } if dup_statelocal_id.present?
+
+    # if first_name or last_name is null skip duplicate detection
+    return { is_duplicate: duplicate_field_data.any?, duplicate_field_data: duplicate_field_data } if patient[:first_name].nil? || patient[:last_name].nil?
+
+    fn_ln_sex_dob_matches = 0
+    fn_ln_sex_matches = 0
+    fn_ln_dob_matches = 0
+    fn_ln_matches = 0
+
+    # The EPI-X format sends this as a Date while Exposure sends it as a string
+    date_of_birth = patient[:date_of_birth].instance_of?(Date) ? patient[:date_of_birth]&.strftime('%F') : patient[:date_of_birth]
+
+    # Determine which type of duplicate exists
+    where(first_name: patient[:first_name], last_name: patient[:last_name])
+      .select('sex', "DATE_FORMAT(date_of_birth, '%Y-%m-%d') date_of_birth").each do |found|
```

Comment:
I would recommend sticking with `.pluck` instead of `.select` since it's more performant

Reply:
I benchmarked this case and pluck was 8x slower

Reply:
resolving this because @Bialogs has determined that the performance is more or less the same after more benchmarking